src/scripts/main.js

```diff
@@ -1,3 +1,42 @@
 'use strict';
 
-// write code here
+const employeeList = document.querySelector('ul');
+
+const parseSalary = (salaryStr) => {
+  let cleanStr = salaryStr.split('$')[1];
+
+  cleanStr = cleanStr.split(',').join('');
+
+  return parseFloat(cleanStr);
+};
+
+const sortEmployees = (list) => {
+  const employeeArray = Array.from(list.children);
+
+  const sortedEmployees = employeeArray.sort((a, b) => {
+    const salaryA = parseSalary(a.getAttribute('data-salary'));
+    const salaryB = parseSalary(b.getAttribute('data-salary'));
```

Comment:
Make sure that the 'data-salary' attribute is present and correctly formatted in the HTML elements. If there's a chance of it being absent or malformed, consider adding checks or default values.